tests/Functional/RsyncFileCopierTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Tests\Functional;
+
+use PhpTuf\ComposerStager\Exception\DirectoryNotFoundException;
+use PhpTuf\ComposerStager\Infrastructure\Process\FileCopier\RsyncFileCopier;
+use Symfony\Component\Process\ExecutableFinder;
+
+/**
+ * @coversDefaultClass \PhpTuf\ComposerStager\Infrastructure\Process\FileCopier\RsyncFileCopier
+ * @covers ::__construct
+ * @covers ::copy
+ * @uses \PhpTuf\ComposerStager\Infrastructure\Filesystem\Filesystem
+ * @uses \PhpTuf\ComposerStager\Infrastructure\Process\ExecutableFinder
+ * @uses \PhpTuf\ComposerStager\Infrastructure\Process\ProcessFactory
+ * @uses \PhpTuf\ComposerStager\Infrastructure\Process\Runner\AbstractRunner
+ */
+class RsyncFileCopierTest extends TestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        if (!self::isRsyncAvailable()) {
+            return;
+        }
+        self::createTestEnvironment();
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        if (!self::isRsyncAvailable()) {
+            return;
+        }
+        self::removeTestEnvironment();
+    }
+
+    protected function setUp(): void
+    {
+        if (!self::isRsyncAvailable()) {
+            $this->markTestSkipped('Rsync is not available for testing.');
+        }
+    }
+
+    protected static function isRsyncAvailable(): bool
+    {
+        $finder = new ExecutableFinder();
+        return !($finder->find('rsync') === null);
+    }
+
+    private function createSut(): RsyncFileCopier
+    {
+        $container = self::getContainer();
+
+        /** @var RsyncFileCopier $sut */
+        $sut = $container->get(RsyncFileCopier::class);
+        return $sut;
+    }
+
+    public function testCopy(): void
+    {
+        touch(self::ACTIVE_DIR . '/lorem.txt');
+        touch(self::ACTIVE_DIR . '/ipsum.txt');
+        touch(self::ACTIVE_DIR . '/dolor-exclude.txt');
+        mkdir(self::ACTIVE_DIR . '/sit-exclude');
```

Comment:
`touch()` and `mkdir()` return TRUE on success and FALSE on failure, so would it maybe make sense to wrap these calls in `self::assertTrue()`?

Reply:
Sure. If this fail so would the rest of the test, but some assertions here would probably help with defect localization in that case.